azure_functions_worker/bindings/meta.py

```diff
@@ -94,5 +107,54 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
             f'unable to encode outgoing TypedData: '
             f'unsupported type "{binding}" for '
             f'Python type "{type(obj).__name__}"')
+    return datum
 
-    return datumdef.datum_as_proto(datum)
+
+def to_outgoing_proto(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type],
+                      shmem_mgr: SharedMemoryManager,
+                      invocation_id: str) -> protos.TypedData:
+    datum = get_datum(binding, obj, pytype)
+    return datumdef.datum_as_proto(datum, shmem_mgr, invocation_id)
+
+
+def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
+                      pytype: typing.Optional[type],
+                      out_name: str,
+                      shmem_mgr: SharedMemoryManager,
+                      invocation_id: str) -> protos.ParameterBinding:
+    datum = get_datum(binding, obj, pytype)
+    # TODO gochaudh: IMPORTANT: Right now we set the AppSetting to disable this
+    # However that takes impact only for data coming from host -> worker
+    # Is there a way to check the AppSetting here so that this does not respond back
+    # with shared memory?
+    param_binding = None
+    if shmem_mgr.is_enabled() and shmem_mgr.is_supported(datum):
+        if datum.type == 'bytes':
```

Comment:
Don't we have the option for string?
We could potentially also support the current stream too.